packages/vscode-extension/src/debugging/RadonCDPProxyDelegate.ts

```diff
@@ -246,7 +255,43 @@ export class RadonCDPProxyDelegate implements CDPProxyDelegate {
     throw new Error("Source map URL schemas other than `data` and `http` are not supported");
   }
 
-  private async handleScriptParsed(command: IProtocolCommand): Promise<IProtocolCommand> {
+  private async setupRadonConnectRuntime(tunnel: ProxyTunnel) {
+    // load script from lib/connect_runtime.js and evaluate it
+    const runtimeScriptPath = path.join(
+      extensionContext.extensionPath,
+      "dist",
+      "connect_runtime.js"
+    );
+    const runtimeScript = fs.readFileSync(runtimeScriptPath, "utf8");
```

Comment:
1. why not the `fs.promises` API? We're already in an `async` context, I don't think we need to stop the whole extension host for this.
2. I dislike the idea of doing fs operations in the `RadonProxyDelegate`, but I'm not sure there's a better way to do it without changing too many things just for the sake of it.
Maybe instead of passing a `boolean` `installConnectRuntime` flag we could pass a `injectedRuntime: string` to `RadonProxyDelegate` instead? We could also then store it in a static somewhere (lazily, after the first read) so that we don't have to read it from the filesystem on each debug connection.

Reply:
I changed this to async, but don't think it is worth optimizing the 2nd point. The RadonCDPProxyDelegate is currently coupled with the new communication mechanism which isn't limited to reading and evaluating the script at a given moment but also to registering the binding. If I was to extract the script loading to uncouple it, I'd also need to extract setting up the binding at the correct moment and also abstract the binding call handlers, which would create an unnecessarily complex abstraction. 

We may want to add a caching mechanism, but given this is only called once per app reload, and that the runtime file will be <1MB I don't think that's necessary either.

Reply:
The difference between how this is handled vs. the runtime binding is, the way we handle the binding here is not really Connect-related (from the perspective of ProxyDelegate) -- you install a binding and emit an event whenever it's called, the mechanism itself is generic and can be reused by different code which wants to talk to the runtime.
At the end of the day, I'd say it's fine either way, it just bugs me a little -- but there are bigger transgressions in the debugger code anyway.